tiledb/dense_array.py

```diff
@@ -441,7 +441,7 @@ def _setitem_impl(self, selection, val, nullmaps: dict):
             labels = {
                 name: (
                     data
-                    if not type(data) is np.ndarray or data.dtype is np.dtype("O")
+                    if not type(data) is np.ndarray or data.dtype == np.dtype("O")
```

Comment:
I forgot to bring this up in the issue, but is there a reason that `tiledb-py` is doing `if type(attr_val) is np.ndarray` instead of `isinstance(attr_val, np.ndarray)`? I think this could present a similar pitfall.